### PR TITLE
Feature: adding oauth2 login endpoint

### PR DESCRIPTION
**Pull request by eapg**
* Adding oauth2 login endpoint

### CODE REVIEW

File: src/lib/repositories/impl_v2/oauth2_repository_impl.py [hunk 1]
```diff
@@ -71,6 +74,19 @@ def _login(
                 client, client_scopes, self.env_config.oauth2_secret_key
             )
 
+            token_decoded = jwt.decode(
```
Comment (reynoldmorel) [inline src/lib/repositories/impl_v2/oauth2_repository_impl.py:77]: Rename to `decoded_token`

File: src/lib/repositories/impl_v2/oauth2_repository_impl.py [hunk 2]
```diff
@@ -82,13 +98,31 @@ def _login(
             refresh_token = build_user_credential_refresh_token(
                 user, client, self.env_config.oauth2_secret_key, client_scopes
             )
+            token_decoded = jwt.decode(
```
Comment (reynoldmorel) [inline src/lib/repositories/impl_v2/oauth2_repository_impl.py:101]: The code is almost the same one as the above, could you optimize this?

File: src/lib/repositories/impl_v2/oauth2_repository_impl.py [hunk 3]
```diff
@@ -82,13 +98,31 @@ def _login(
...
+                access_token, self.env_config.oauth2_secret_key, algorithms="HS256"
+            )
+
+            authentication_response = {
+                "user": {
```
Comment (reynoldmorel) [inline src/lib/repositories/impl_v2/oauth2_repository_impl.py:106]: why not just `"user": decoded_token.get('user')`?

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/lib/repositories/impl_v2/oauth2_repository_impl.py (hunk 1 above)
Comment (eapg) [reply src/lib/repositories/impl_v2/oauth2_repository_impl.py]: done.

File: src/lib/repositories/impl_v2/oauth2_repository_impl.py (hunk 2 above)
Comment (eapg) [reply src/lib/repositories/impl_v2/oauth2_repository_impl.py]: done.

File: src/lib/repositories/impl_v2/oauth2_repository_impl.py (hunk 3 above)
Comment (eapg) [reply src/lib/repositories/impl_v2/oauth2_repository_impl.py]: done.

File: src/utils/oauth2_util.py [hunk 4]
```diff
@@ -445,3 +454,20 @@ def is_endpoint_protected(endpoint_request):
...
+    decoded_token = jwt.decode(access_token, secret_key, algorithms="HS256")
+
+    authentication_response = {
+        "access_token": access_token,
+        "expires_in": get_time_in_seconds_from_unix_time(decoded_token.get("exp")),
+        "refresh_token": refresh_token,
+        "scopes": decoded_token.get("scopes"),
+        "client_name": decoded_token.get("client_name"),
+    }
+    if decoded_token.get("user"):
```
Comment (reynoldmorel) [inline src/utils/oauth2_util.py:470]:
```
user = decoded_token.get("user")

if user:
  authentication_response.update({"user": user})

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/utils/oauth2_util.py (hunk 4 above)
Comment (eapg) [reply src/utils/oauth2_util.py]: done.

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**